### PR TITLE
HDDS-8964. Create a separate endpoint for summary for various SCM Insights and OM Insights.

### PR DESCRIPTION
**Pull request by ArafatKhan2198**
## What changes were proposed in this pull request?
In summary, this patch introduces separate API endpoints for retrieving summarized information about open keys and deleted keys in Hadoop Ozone Recon. These summaries provide aggregated statistics and high-level information about the keys' status and data sizes.
Changes made :- 
- `OMDBInsightEndpoint.java` : This file adds two new API endpoints: `/open/summary` and `/deletePending/summary`. These endpoints retrieve summaries of open keys and deleted keys, providing aggregated information like total key count, replicated data size, and unreplicated data size.
- `KeyInsightInfoResponse.java` : This Response class no longer includes the `keysSummary` field, as the keys' summary information is now included directly in the response body of the API endpoints.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8964

## How was this patch tested?

Ran the Existing UT's and also tested manually 

image
image

### CODE REVIEW

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java [hunk 1]
```diff
@@ -255,15 +253,39 @@ public Response getOpenKeyInfo(
         break;
       }
     }
-    // Populate the keysSummary map
-    createKeysSummaryForOpenKey(keysSummary);
-
-    openKeyInsightInfo.setKeysSummary(keysSummary);
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java:261]: @ArafatKhan2198 Thanks for working on this patch, If we have removed keySummary from this API endpoint, then kindly update API doc as well.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java [hunk 2]
```diff
@@ -364,6 +380,33 @@ private void getPendingForDeletionKeyInfo(
...
+   */
+  @GET
+  @Path("/deletePending/summary")
+  public Response getDeletedKeySummary() {
+    // Create a HashMap for the keysSummary
+    Map<String, Long> keysSummary = new HashMap<>();
+    // Create a keys summary for deleted keys
+    createKeysSummaryForDeletedKey(keysSummary);
+    return Response.ok(keysSummary).build();
+  }
```
Comment (devmadhuu) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java:408]: Same comment, pls update the API doc and remove summary part.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java (hunk 1 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java:261]: Thanks for the comment @devmadhuu 
I have made the changes!

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java (hunk 2 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java:403]: Done!

**Comment by ashishkumar50** [comment]
@ArafatKhan2198 Thanks for working on this, LGTM +1

**Comment by smitajoshi12** [comment]
Looks good to me

**Review by dombizita — APPROVED** [review]
thanks for working on this @ArafatKhan2198, it looks good to me!

**Comment by dombizita** [comment]
thanks for the patch @ArafatKhan2198! thanks for the review @devmadhuu, @ashishkumar50 and @smitajoshi12!

**Comment by dombizita** [comment]
after this merge there are failures in `master`, check [here](https://github.com/apache/ozone/actions/runs/5853773657/job/15868438276), I'm looking into it now

**Comment by dombizita** [comment]
I raised #5185 to fix it.

**Comment by dombizita** [comment]
#5185 is now merged, thanks for the quick review and merge @sodonnel!

**Pull request merged**